### DISCUSSION TITLE
After closing the tooltip by using ESC or the close button, when I restart the tour, how do I start from the begining?

### DISCUSSION THREAD

**Discussion by jiyang-sas** [gilbarbara/react-joyride#656] [Q&A]
It seems the current behavior is continue from the last tour step number. Thanks!
I am using a custom tooltip component and version 2.2.1. 

Thanks!

**Comment by jiyang-sas** [comment]
I tried to manage the state myself by setting  stepIndex and now I cannot restart the tour any more once it is closed (by ESC, close button).  Any idea?

const [state, setState] = useState({ run: true, stepIndex: 0 });

  // override the callback
  const reactCallback =  (data: CallBackProps) => {
    const { action, index, type, status } = data;
    if (
      ([STATUS.FINISHED, STATUS.SKIPPED] as string[]).includes(status) ||
      action === ACTIONS.CLOSE
    ) {
      // Need to set our running state to false, so we can restart if we click start again.
      setState({ run: false, stepIndex: 0 });
    } else if (([EVENTS.STEP_AFTER, EVENTS.TARGET_NOT_FOUND] as string[]).includes(type)) {
      const stepIndex = index + (action === ACTIONS.PREV ? -1 : 1);

      setState({ ...state, stepIndex: stepIndex });
    }
  };

**Reply by venumanthena** [reply]
add this this.setState({ run: true, stepIndex: 0 }) on your button click at start.

**Reply by jiyang-sas** [reply]
I did have that one, but it did not work. Thx
